### ISSUE TITLE
Remove streamid

### ISSUE THREAD

**Issue by rejexy** [Edward-Wu/srt-live-server#52]
Due to problems in vlc is there a way to create a broadcast without a streamid?
For example, if this parameter is not set create **localhost:8080/live**.

**Comment by rejexy** [comment]
It would also be great if there was access to the broadcast via the http protocol.

**Comment by wonder-wanderer** [comment]
**We also need to send and receive a stream just on the UDP port, without any StreamID (well optionally with a passphrase)**.

**This is very important because we also need to use SRT with VLC and many other products that DO NOT support Stream ID**. And if one of them finally begins to support stream ID, many others still don't! 

So it is a shame that must abandon using this great SRT server just because it requires StreamID. Or is there a way to not require stream ID?

VLC will be soon have support for stream ID as mentioned here:

- https://github.com/Haivision/srt/issues/1015
- https://forum.videolan.org/viewtopic.php?f=7&p=506009

but many solutions still don't.

**Comment by jengajenga** [comment]
> **We also need to send and receive a stream just on the UDP port, without any StreamID (well optionally with a passphrase)**.
> 
> **This is very important because we also need to use SRT with VLC and many other products that DO NOT support Stream ID**. And if one of them finally begins to support stream ID, many others still don't!
> 
> So it is a shame that must abandon using this great SRT server just because it requires StreamID. Or is there a way to not require stream ID?
> 
> VLC will be soon have support for stream ID as mentioned here:
> 
> * [Haivision/srt#1015](https://github.com/Haivision/srt/issues/1015)
> * https://forum.videolan.org/viewtopic.php?f=7&p=506009
> 
> but many solutions still don't.

vmix can receive.

**Comment by whyronimus** [comment]
I can't receive any SRT streams with nightly build 4.0 of VLC, though. 

SLS output (numbers as X stand for my IP) when I try to connect with VLC:

> 2020-12-14 22:14:52:577 SLS INFO: [0x55e5f681be00]CSLSSrt::libsrt_accept ok, new sock=32342367, XX.XXX.XXX.XXX:63001.
> 2020-12-14 22:14:52:577 SLS INFO: [0x7fa83c0b2010]CSLSListener::handler, new client[XX.XXX.XXX.XXX:63001], fd=32342367.
> 2020-12-14 22:14:52:577 SLS INFO: [0x7fa83c0b2010]CSLSListener::handler, [XX.XXX.XXX.XXX:63001], sid 'url.to.server/srt_live/stream'
> 2020-12-14 22:14:52:577 SLS ERROR: [0x7fa83c0b2010]CSLSListener::handler, refused, new role[XX.XXX.XXX.XXX:63001], stream='url.to.server/srt_live/stream',but publisher=0x7fa83a827010 is not NULL.
> 2020-12-14 22:14:52:577 SLS INFO: [0x7fa834002710]CSLSSrt::libsrt_close, fd=32342367.

Is there anything I did wrong?

**Comment by wonder-wanderer** [comment]
> vmix can receive.

That's true, but like I wrote the whole point is that many sollutions (SRT products) still can't.

I am sending and receiving with many people using various SRT products. So I need to connect SRT with more and more different SRT products and unfortunately it looks like the situation for many at least months is going to be that many SRT products will not support StreamID.

**Comment by wonder-wanderer** [comment]
**... so is there a way to not require stream ID? or is it in a roadmap?**

**Comment by jengajenga** [comment]
> **... so is there a way to not require stream ID? or is it in a roadmap?**

It looks like you may have to get a coder to do this for you.

**Comment by wonder-wanderer** [comment]
I am asking is there a way within srt-live-server **to make srt-live-server not require stream ID** (for input and output, i.e. an app or config that has only a UDP port but no input output domain, stream ID, etc)? or is it in srt-live-server roadmap?

@Edward-Wu could you give any hints? Thanks in advance! Much appreciated!

**Comment by wonder-wanderer** [comment]
@Edward-Wu or anybody actually - is this feature (of not requiring StreamID) in any roadmap, or work?
